PR: https://github.com/odoo/odoo/pull/161234

From: 940fe70b3c64544d4e0cb96ee754d1534a2a44b9
From: taal-odoo
Date: 2024-08-21 10:42:04

Breaking data model changes scores: del:3 + add:0, change matches:
-    gift_card_settings = fields.Selection(
-    _inherit = 'res.config.settings'
-    pos_gift_card_settings = fields.Selection(related='pos_config_id.gift_card_settings', readonly=False)

Total Changes: 30

[IMP] pos_loyalty: sell physical giftcard from terminal

Before the commit:
*--
Gift card codes were generated automatically, without any option for manual
input or selling physical gift cards.

After the commit:
*--
* Added the "Sell Physical Gift Card" option under gift card products
in POS.
* Allows for the manual input of gift card numbers.
* Orders generate gift cards with entered numbers.
* Existing gift cards can have amounts added.

Related PR:
* Upgrade: odoo/upgrade#6363

task-3647760

closes odoo/odoo#161234

Signed-off-by: Joseph Caburnay (jcb) <jcb@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_loyalty/models/__init__.py
+++ b/addons/pos_loyalty/models/__init__.py
@@ -11,6 +11,5 @@ from . import pos_config
 from . import pos_order_line
 from . import pos_order
 from . import pos_session
-from . import res_config_settings
 from . import product_product
 from . import res_partner

--- a/addons/pos_loyalty/models/pos_config.py
+++ b/addons/pos_loyalty/models/pos_config.py
@@ -7,15 +7,6 @@ from odoo.exceptions import UserError
 class PosConfig(models.Model):
     _inherit = 'pos.config'
 
-    gift_card_settings = fields.Selection(
-        [
-            ("create_set", "Generate PDF cards"),
-            ("scan_use", "Scan existing cards"),
-        ],
-        string="Gift Cards settings",
-        default="create_set",
-        help="Defines the way you want to set your gift cards.",
-    )
     # NOTE: this funtions acts as a m2m field with loyalty.program model. We do this to handle an excpetional use case:
     # When no PoS is specified at a loyalty program form, this program is applied to every PoS (instead of none)
     def _get_program_ids(self):
@@ -63,11 +54,10 @@ class PosConfig(models.Model):
                 reward = gc_program.reward_ids
                 if reward.reward_type != 'discount' or reward.discount_mode != 'per_point' or reward.discount != 1:
                     raise UserError(_('Invalid gift card program reward. Use 1 currency per point discount.'))
-                if self.gift_card_settings == "create_set":
-                    if not gc_program.mail_template_id:
-                        raise UserError(_('There is no email template on the gift card program and your pos is set to print them.'))
-                    if not gc_program.pos_report_print_id:
-                        raise UserError(_('There is no print report on the gift card program and your pos is set to print them.'))
+                if not gc_program.mail_template_id:
+                    raise UserError(_('There is no email template on the gift card program and your pos is set to print them.'))
+                if not gc_program.pos_report_print_id:
+                    raise UserError(_('There is no print report on the gift card program and your pos is set to print them.'))
 
         return super()._check_before_creating_new_session()
 

--- a/addons/pos_loyalty/models/pos_order.py
+++ b/addons/pos_loyalty/models/pos_order.py
@@ -73,7 +73,7 @@ class PosOrder(models.Model):
         coupon_create_vals = [{
             'program_id': p['program_id'],
             'partner_id': get_partner_id(p.get('partner_id', False)),
-            'code': p.get('barcode') or self.env['loyalty.card']._generate_code(),
+            'code': p.get('code') or p.get('barcode') or self.env['loyalty.card']._generate_code(),
             'points': 0,
             'source_pos_order_id': self.id,
         } for p in coupons_to_create.values()]

--- a/addons/pos_loyalty/models/res_config_settings.py
+++ b/None
@@ -1,9 +0,0 @@
-# -*- coding: utf-8 -*-
-
-from odoo import fields, models, api
-
-
-class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
-
-    pos_gift_card_settings = fields.Selection(related='pos_config_id.gift_card_settings', readonly=False)
